src/machine/machinectl.c

```diff
@@ -487,14 +533,21 @@ static int print_addresses(sd_bus *bus, const char *name, int ifi, const char *p
         if (r < 0)
                 return r;
 
+        addresses = strdup(prefix);
+        if (!addresses)
+                return log_oom();
+        prefix = "";
+
         r = sd_bus_message_enter_container(reply, 'a', "(iay)");
         if (r < 0)
                 return bus_log_parse_error(r);
 
         while ((r = sd_bus_message_enter_container(reply, 'r', "iay")) > 0) {
                 int family;
                 const void *a;
+                char *tmp;
                 size_t sz;
+                char buf_ifi[24];
                 char buffer[MAX(INET6_ADDRSTRLEN, INET_ADDRSTRLEN)];
```

Comment:
Note that it matters much, but please consider placing the "char" allocations in one line:

```
char *tmp, buf_ifi[…], buffer[…];
```